src/commands/command-lib/mod/streammessage.cmdlib.ts

```diff
@@ -0,0 +1,158 @@
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see <https://www.gnu.org/licenses/>.
+ */
+
+import * as Discord from "discord.js";
+import { DrBotGlobal } from "@src/interfaces/global.js";
+import moment from "moment-timezone";
+import storage from "@src/lib/utilities/storage.js";
+import { DrBotSlashCommand } from "@src/lib/base/DrBotCommand.js";
+import { DrBotSubcommand } from "@src/lib/base/DrBotSubcommand.js";
+
+declare const global: DrBotGlobal;
+
+export default class ModStreamMessage extends DrBotSubcommand {
+    static parentCommand: string = "Mod";
+
+    public async setup(
+        parentSlashCommand: Discord.SlashCommandBuilder
+    ): Promise<boolean> {
+        parentSlashCommand.addSubcommand((subcommand) =>
+            subcommand
+                .setName("streammessage")
+                .setDescription("Set a stream message for a role")
+                .addStringOption((option) =>
+                    option
+                        .setName("message")
+                        .setDescription("Stream Message")
+                        .setRequired(true)
+                )
+                .addStringOption((option) =>
+                    option
+                        .setName("time")
+                        .setDescription("Time for the stream message")
+                        .setRequired(true)
+                )
+                .addRoleOption((option) =>
+                    option.setName("role").setDescription("Select a role")
+                )
+                .addStringOption((option) =>
+                    option
+                        .setName("template")
+                        .setDescription("Select a template")
+                        .addChoices({
+                            name: "stream", value: "stream"
+                        }, {
+                            name:"cancellation", value: "cancellation"})
+                )
+        );
+        this._loaded = true;
+        return true;
+    }
+
+    public async runSubCommand(interaction: Discord.CommandInteraction) {
+        await interaction.deferReply({
+            ephemeral: true,
+        });
+
+        //Get Channel List
+        const channels = await interaction.client.guilds.fetch(global.app.config.mainServer).then(guild => guild.channels.fetch())
+        // check if there is a channel that includes "birthdays" in it's name
+        let channel;
+        if (!channels.some((channel) => /live-(streams|notifications|notifs|announcements)/i.test(channel.name) && channel.type == Discord.ChannelType.GuildText)) {
+
+            if (!channels.some((channel) => /announcements/i.test(channel.name) && channel.type == Discord.ChannelType.GuildText)) {
+                interaction.editReply({content: "No channel found with the name `live-streams`, `live-notifications`, `live-notifs`, `live-announcements`, or `announcements`"});
```

Comment:
I think we could make this a bit more user-friendly but also a bit more professional and PoV-like (I'm not sure if my suggestion will work as expected but if not, you can tweak it so it works)
```suggestion
                interaction.editReply({content: "I was unable to find a valid channel for the stream announcement.\n#- Searched for: `live-streams`, `live-notifications`, `live-notifs`, `live-announcements`, and `announcements`"});
```

Reply:
I understand this.... I think I forget that its a chatbot interacting with users not a CLI application